src/Core__Float.resi

```diff
@@ -1,26 +1,404 @@
+
+  ## Examples
+
+  ```rescript
+  Float.Constants.positiveInfinity
+  ```
+  */
+  @val
+  external positiveInfinity: float = "Number.POSITIVE_INFINITY"
+
+  /**
+  The negative Infinity value
+  See [`Number.NEGATIVE_INFINITY`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/NEGATIVE_INFINITY) on MDN.
+
+  ## Examples
+
+  ```rescript
+  Float.Constants.negativeInfinity
+  ```
+  */
+  @val
+  external negativeInfinity: float = "Number.NEGATIVE_INFINITY"
+
+  /**
+  The smallest positive numeric value representable in JavaScript.
+  See [`Number.MIN_VALUE`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MIN_VALUE) on MDN.
+
+  ## Examples
+
+  ```rescript
+  Float.Constants.minValue
+  ```
+  */
+  @val
+  external minValue: float = "Number.MIN_VALUE"
+
+  /**
+  The maximum positive numeric value representable in JavaScript.
+  See [`Number.MAX_VALUE`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_VALUE) on MDN.
+
+  ## Examples
+
+  ```rescript
+  Float.Constants.minValue
+  ```
+  */
+  @val
+  external maxValue: float = "Number.MAX_VALUE"
 }
-@val external isNaN: float => bool = "isNaN"
-@val external isFinite: float => bool = "isFinite"
-@val external parseFloat: 'a => float = "parseFloat"
-@val external parseInt: 'a => float = "parseInt"
-@val external parseIntWithRadix: ('a, ~radix: int) => float = "parseInt"
-@send external toExponential: float => string = "toExponential"
-@send external toExponentialWithPrecision: (float, ~digits: int) => string = "toExponential"
-@send external toFixed: float => string = "toFixed"
-@send external toFixedWithPrecision: (float, ~digits: int) => string = "toFixed"
-@send external toPrecision: float => string = "toPrecision"
-@send external toPrecisionWithPrecision: (float, ~digits: int) => string = "toPrecision"
-@send external toString: float => string = "toString"
-@send external toStringWithRadix: (float, ~radix: int) => string = "toString"
-@send external toLocaleString: float => string = "toLocaleString"
+
+/**
+`isNaN(v)` tests if the given `v` is `NaN`.
+See [`NaN`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/NaN) on MDN.
+
+## Examples
+
+```rescript
+Float.isNaN(3.0) // false
+Float.isNaN(Float.Constants.nan) // true
+```
+*/
+@val
+external isNaN: float => bool = "isNaN"
+
+/**
+`isFinite(v)` tests if the given `v` is finite.
+See [`isFinite`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isFinite) on MDN.
+
+## Examples
+
+```rescript
+Float.isFinite(1.0) // true
+Float.isFinite(Float.Constants.nan) // false
+Float.isFinite(Float.Constants.positiveInfinity) // false
+```
+*/
+@val
+external isFinite: float => bool = "isFinite"
+
+/**
+`parseFloat(v)` parse the given `v` and returns a float. Leading whitespace in
```

Comment:
I think we should add a note here that Float.fromString can be used if you want to ensure you get a valid float back and not NaN.

Reply:
Both should also explain what the expected format is. Consumers might expect it to be locale-aware, or to be able to parse rescript float literals. The expected format is that of JS number literals.

Reply:
> I think we should add a note here that Float.fromString can be used if you want to ensure you get a valid float back and not NaN.

For this, we must change the signature of `parseFloat` to `string => float`

The signature is currently `'a => float`

Reply:
Done https://github.com/rescript-association/rescript-core/pull/54/commits/e003d749ab51126a90be31200398296115a2349f